SLC-S-GQIMonitor/SLC-AS-GQIMonitor.VersionCheck/SLC-AS-GQIMonitor.VersionCheck.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using Skyline.DataMiner.Automation;
+using Skyline.DataMiner.Net;
+
+namespace SLCASGQIMonitorVersionCheck
+{
+	/// <summary>
+	/// Represents a DataMiner Automation script.
+	/// </summary>
+	public sealed class Script
+	{
+		/// <summary>
+		/// The script entry point.
+		/// </summary>
+		/// <param name="engine">Link with SLAutomation process.</param>
+		public void Run(IEngine engine)
+		{
+			try
+			{
+				RunSafe(engine);
+			}
+			catch (ScriptAbortException)
+			{
+				// Catch normal abort exceptions (engine.ExitFail or engine.ExitSuccess)
+				throw; // Comment if it should be treated as a normal exit of the script.
+			}
+			catch (ScriptForceAbortException)
+			{
+				// Catch forced abort exceptions, caused via external maintenance messages.
+				throw;
+			}
+			catch (ScriptTimeoutException)
+			{
+				// Catch timeout exceptions for when a script has been running for too long.
+				throw;
+			}
+			catch (InteractiveUserDetachedException)
+			{
+				// Catch a user detaching from the interactive script by closing the window.
+				// Only applicable for interactive scripts, can be removed for non-interactive scripts.
+				throw;
+			}
+			catch (Exception e)
+			{
+				engine.ExitFail("Run|Something went wrong: " + e);
+			}
+		}
+
+		private string GetGQIVersion(IConnection connection)
+		{
+			var versionFetcher = new GQIMonitor.VersionFetcher();
+			return versionFetcher.GetGQIVersion(connection);
+		}
+
+		private void RunSafe(IEngine engine)
+		{
+			var requiredVersionParam = engine.GetScriptParam(2);
+
+			if (requiredVersionParam is null || string.IsNullOrEmpty(requiredVersionParam.Value))
+				return;
+
+			var requiredVersion = requiredVersionParam.Value;
+			var actualVersion = GetGQIVersion(engine.GetUserConnection());
+
+			if (IsVersionGreaterOrEqual(actualVersion, requiredVersion))
+				return;
+
+			var webAppVersionParam = engine.GetScriptParam(3);
+
+			if (webAppVersionParam is null || string.IsNullOrEmpty(webAppVersionParam.Value))
+				return;
+
+			UIBuilder uiBuilder = new UIBuilder
+			{
+				RequireResponse = true,
+				RowDefs = "a;a",
+				ColumnDefs = "a",
+			};
+
+			uiBuilder.Title = "Incorrect GQI Version";
+
+			UIBlockDefinition blockStaticText = new UIBlockDefinition();
+			blockStaticText.Type = UIBlockType.StaticText;
+			blockStaticText.Text = $"This page requires DataMiner web app version {webAppVersionParam.Value}.";
```

Comment:
Indicate that any higher version is also ok?

```suggestion
			blockStaticText.Text = $"This page requires DataMiner web app version {webAppVersionParam.Value} or newer.";
```

Should we also indicate that it requires the GQI DxM?